src/main/java/dbaccessor/user/UserAccessor.java

```diff
@@ -189,4 +189,25 @@ public User getUser(long userId) {
         Statement statement = Statement.newBuilder(sqlStatment).bind("userId").to(userId).bind("listOfChatIdOfUser").toInt64Array(listOfChatIdOfUser).build();
         return ImmutableList.copyOf(spannerTemplate.query(ListChats.UsernameChatId.class, statement,  new SpannerQueryOptions().setAllowPartialRead(true)));
     }
+
+    /**
+     * Given the userId of a user A and a string mobileNoPrefix, returns all users such that either of the following is true:
+     *  1. There is a Chat between user A and this user and the mobile number of this user starts with the string mobileNoPrefix.
+     *  2. MobileNo of this user is equal to mobileNoPrefix.
+     */
+    public ImmutableList<User> getUsersByMobileNumber(long userId, String mobileNoPrefix) {
+        String sqlStatement = "SELECT * from User where MobileNo LIKE @mobileNoPrefix AND " +
```

Comment:
Should I build index on MobileNo, this might make `SELECT * FROM User WHERE MobileNo=@mobileNo` a bit faster.

Reply:
It would be better actually if its not too much effort. Otherwise we can probably leave it for now, after entering 1 million entries, what is the latency for phone number api(without index)?

Reply:
Total time that it is taking for getting the response for phone number API is around 1 to 2 seconds currently (this includes the network latency).

Reply:
Yeah. We should have an index.
That should improve the performance by a lot.

Reply:
Added.